### PR TITLE
add options to reveal methods of `TextEditor`

### PR DESCRIPTION
**Pull request by AlexTugarev**
This change makes it possible to reveal a line, or a range at the top of editor, which is a prerequisite for PR #1051.

### CODE REVIEW

**Comment by AlexTugarev** [comment]
@svenefftinge, do you mean like this https://github.com/theia-ide/theia/pull/1079/commits/799805513f0423f8c908b0affd403f987b7e3afa#diff-54c26fcbd101d4ec3adffa9152ace13eR43 ?

```
revealRange(range: Range, revealAt?: VerticalRevealType): void;
```
and
```
export const enum VerticalRevealType {
    Default = 0,
    Top = 1,
    Center = 2,
    CenterIfOutside = 3,
}
```

**Comment by AlexTugarev** [comment]
BTW I'll update and clean up the branch in end.

File: packages/editor/src/browser/editor.ts [hunk 1]
```diff
@@ -40,7 +40,8 @@ export interface TextEditor extends Disposable, TextEditorSelection {
     readonly onFocusChanged: Event<boolean>;
 
     revealPosition(position: Position): void;
-    revealRange(range: Range): void;
+    revealRange(range: Range, revealAt?: VerticalRevealType): void;
```
Comment (akosyakov) [inline packages/editor/src/browser/editor.ts]: Could we use options pattern as at other places? It allows adding new options later, like to specify how a range should be revealed horizontally.
```ts
/**
 * by default a range is revealed vertically at center
 */
revealRange(range: range, options?: RevealRangeOptions): void;

interface RevealRangeOptions {
    at: 'center' | 'top' | 'centerIfOutsideViewport'
}
```
Also i find it is quite readable at the client side:
```ts
editor.reveal(range, { at: 'top' })
```

File: packages/editor/src/browser/editor.ts [hunk 2]
```diff
@@ -40,7 +40,8 @@ export interface TextEditor extends Disposable, TextEditorSelection {
     readonly onFocusChanged: Event<boolean>;
 
     revealPosition(position: Position): void;
```
Comment (akosyakov) [inline packages/editor/src/browser/editor.ts]: Should we add options to `revealPosition` as well? fyi: looking at monaco API, one can specify whether a position should be revealed vertically or horizontally.

File: packages/editor/src/browser/editor.ts (hunk 1 above)
Comment (AlexTugarev) [reply packages/editor/src/browser/editor.ts]: sounds good!

Comment (AlexTugarev) [reply packages/editor/src/browser/editor.ts]: I've updated it.

File: packages/editor/src/browser/editor.ts (hunk 2 above)
Comment (AlexTugarev) [reply packages/editor/src/browser/editor.ts]: Please, have a look.

File: packages/monaco/src/browser/monaco-editor.ts [hunk 3]
```diff
@@ -181,14 +183,37 @@ export class MonacoEditor implements TextEditor, IEditorReference {
         return this.onSelectionChangedEmitter.event;
     }
 
-    revealPosition(raw: Position): void {
+    revealPosition(raw: Position, options: RevealPositionOptions = { type: 'verticalInCenter', horizontal: 'reveal' }): void {
         const position = this.p2m.asPosition(raw);
-        this.editor.revealPositionInCenter(position);
+        switch (options.type) {
+            case 'verticalInCenter':
+                this.editor.revealPosition(position, true, options.horizontal === 'reveal');
```
[the remark this answers is not in the document]
Comment (AlexTugarev) [reply packages/monaco/src/browser/monaco-editor.ts]: Please, have a look.

File: packages/monaco/src/browser/monaco-editor.ts [hunk 4]
```diff
@@ -181,14 +183,37 @@ export class MonacoEditor implements TextEditor, IEditorReference {
         return this.onSelectionChangedEmitter.event;
     }
 
-    revealPosition(raw: Position): void {
+    revealPosition(raw: Position, options: RevealPositionOptions = { vertical: 'center', horizontal: 'reveal' }): void {
         const position = this.p2m.asPosition(raw);
-        this.editor.revealPositionInCenter(position);
+        switch (options.vertical) {
+            case 'auto':
+                this.editor.revealPosition(position, false, options.horizontal === 'reveal');
```
Comment (akosyakov) [inline packages/monaco/src/browser/monaco-editor.ts]: looks like `options.horizontal` can be boolean, other than that all cases seems to be covered now

Comment (AlexTugarev) [reply packages/monaco/src/browser/monaco-editor.ts]: ✔️

**Pull request merged**